### PR TITLE
Fix bug in moving files between subfolders.

### LINKED ISSUE #102: Move error when moving between subfolders
Hi! Thanks for this very useful piece of software! I was trying to rename a file in a particular subfolder of my Dropbox:
`dbxcli mv /foo/tmp/trace.pdf /foo/tmp/trace2.pdf -v`

Here's the error output that I got:
```
2018/09/11 19:28:55 sdk.go:93: arg: &{{/foo/tmp/trace.pdf /foo/tmp/trace2.pdf//foo/tmp/trace.pdf} false false false}
2018/09/11 19:28:55 sdk.go:93: req: &{POST https://api.dropboxapi.com/2/files/move_v2 HTTP/1.1 1 1 map[Content-Type:[application/json]] {0xc420079620} 0x125e910 234 [] false api.dropboxapi.com map[] map[] <nil> map[]   <nil> <nil> <nil> <nil>}
2018/09/11 19:28:55 sdk.go:93: resp: &{409 Conflict 409 HTTP/1.1 1 1 map[Connection:[keep-alive] X-Dropbox-Request-Id:[744e8225b8b1c569c19469211a98a92b] X-Frame-Options:[DENY] Date:[Tue, 11 Sep 2018 23:28:55 GMT] Content-Type:[application/json] X-Content-Type-Options:[nosniff] Content-Disposition:[attachment; filename='error'] Server:[nginx] Content-Security-Policy:[sandbox; frame-ancestors 'none']] 0xc420210080 -1 [chunked] false false map[] 0xc4200fe300 0xc42033a000}
2018/09/11 19:28:55 sdk.go:93: body: [123 34 101 114 114 111 114 95 115 117 109 109 97 114 121 34 58 32 34 116 111 47 109 97 108 102 111 114 109 101 100 95 112 97 116 104 47 46 46 34 44 32 34 101 114 114 111 114 34 58 32 123 34 46 116 97 103 34 58 32 34 116 111 34 44 32 34 116 111 34 58 32 123 34 46 116 97 103 34 58 32 34 109 97 108 102 111 114 109 101 100 95 112 97 116 104 34 125 125 125]
Move error: &{{/foo/tmp/trace.pdf /foo/tmp/trace2.pdf//foo/tmp/trace.pdf} false false false}
```

I also tried moving a file from one subfolder to a different, existing subfolder. This results in a similar error: Looking at the first line of the output, it seems that dbxcli appends the full source path to the destination path, and not just the part after the last slash, as GNU mv would do. Or am I using the command in the wrong way?

### LINKED ISSUE #104: Provide credentials thru environment variable?
It is often useful in CI systems etc to provide credentials via an environment variable. Is this possible in dbxcli?

PS: Holy crap @diwakergupta its been a long time and no see! ✋

### PR DESCRIPTION
**Pull request by vaithak**
Fixes #102 , Fixes #104

### CODE REVIEW

**Comment by vaithak** [comment]
This PR fixes two issues - #102 & #104 . Please review this.
Thanks.

**Comment by diwakergupta** [comment]
@vaithak thanks. Can you split this up into separate PRs for each issue? For #102 I'd also recommend adding something to `contrib/test.sh` that validates the behavior. We don't have comprehensive integration tests, but some way to verify before / after would help.

**Comment by vaithak** [comment]
@diwakergupta. There is already a test in `contrib/test.sh` for mv command.  
Can you please elaborate what test should I write in it ?

**Comment by vaithak** [comment]
@diwakergupta I have added tests as well as updated `mv.go` .

**Comment by vaithak** [comment]
@diwakergupta Please review my PR.

File: cmd/mv.go [hunk 1]
```diff
@@ -39,13 +40,28 @@ func mv(cmd *cobra.Command, args []string) error {
...
-		if err != nil {
-			relocationError := fmt.Errorf("Error validating move for %s to %s: %v", argument, destination, err)
-			mvErrors = append(mvErrors, relocationError)
+
+		argumentFile := re.FindString(argument)
+		lastCharDest := destination[len(destination)-1:]
+
+		if lastCharDest == "/" {
+			arg, err := makeRelocationArg(argument, destination + argumentFile)
+			if err != nil {
```
Comment (diwakergupta) [inline cmd/mv.go]: Can you split this out of the if-else block? Lines 51-56 are exactly the same as lines 59-64.

File: contrib/test.sh [hunk 2]
```diff
@@ -35,8 +35,8 @@ ${dbxcli} cp ${d}/dbxcli ${d}/dbxcli-new
 echo "Testing revs"
 rev=$(${dbxcli} revs ${d}/dbxcli)
 
-echo "Testing mv"
-${dbxcli} mv ${d}/dbxcli ${d}/dbxcli-old
+echo "Testing updated mv"
```
Comment (diwakergupta) [inline contrib/test.sh]: Let's just keep the old test and add a new one so we test both the trailing slash and without slash

**Comment by vaithak** [comment]
@diwakergupta I have added the requested changes. Please review them.

File: cmd/mv.go (hunk 1 above)
Comment (vaithak) [reply cmd/mv.go]: ok! I have updated the cmd/mv.go file with requested changes.

File: contrib/test.sh (hunk 2 above)
Comment (vaithak) [reply contrib/test.sh]: Ok! Updated test.sh with both tests.

**Comment by vaithak** [comment]
@diwakergupta I have made all the changes, please accept my PR.

**Pull request merged**